HARK/ConsumptionSaving/ConsRiskyAssetModel.py

```diff
@@ -202,25 +207,11 @@ def get_Adjust(self):
         -------
         None
         """
-        if not ("AdjustPrb" in self.time_vary):
-
-            self.shocks["Adjust"] = Bernoulli(
-                self.AdjustPrb, seed=self.RNG.randint(0, 2 ** 31 - 1)
-            ).draw(self.AgentCount)
-
-        else:
-
-            Adjust = np.zeros(self.AgentCount)  # Initialize shock array
-            for t in range(self.T_cycle):
-                these = t == self.t_cycle
-                N = np.sum(these)
-                if N > 0:
-                    AdjustPrb = self.AdjustPrb[t - 1]
-                    Adjust[these] = Bernoulli(
-                        AdjustPrb, seed=self.RNG.randint(0, 2 ** 31 - 1)
-                    ).draw(N)
-
-            self.shocks["Adjust"] = Adjust
+        self.shocks['Adjust'] = IndexDistribution(
+            Bernoulli,
+            {'p' : self.AdjustPrb},
+            seed=self.RNG.randint(0, 2 ** 31 - 1)
+            ).draw(self.t_cycle)
```

Comment:
@sbenthall thinking about indices more carefully now, I think that according to the previous code and time indexing of solutions vs simulations this should be `.draw(self.t_cycle - 1)`, shouldn't it?

There is still the question of why would newborns use the final parameters (`0-1 = -1`) but that might be a discussion for another time.

Reply:
yes, that's right.

In python, an index of -1 gets you the last element of a list/collection.

so correcting this means working in the extra 'newborn' logic.

I think it would be better to figure out how to fix #1022 across the board.

Reply:
Well yes, newborns are a separate issue.

But the `.draw(self.t_cycle)` is still a bug that should be fixed by turning it into `.draw(self.t_cycle - 1)`, isn't it?

Reply:
Hmm.

Prior to this PR:

```
AdjustPrb = self.AdjustPrb[t - 1]
                    Adjust[these] = Bernoulli(
                        AdjustPrb, seed=self.RNG.randint(0, 2 ** 31 - 1)
                    ).draw(N)
```

For agents of age t, use AdjustPr[t -1]. If t = 0, this would use AdjustPrb[-1], the _last_ AdjustPrb.
This is a bug. It never surfaced because there is no test case for varying AdjustPrbs?

After #1018 merge:

` (IndexDistribution of AdjustPrbs).draw(self.t_cycle)`

Now the draws are "off by one".

Either way, there is a bug! But this was a case that nobody bothered writing a test for.

Whoever wrote the original AdjustPrb simulation code forgot to resolve the newborn case.

Now the question is: should we put the special newborn case in everywhere that it's been left out?

My preference is to fix #1022 instead. But your point is well-taken that this is a problem with the implementation at the moment. I'll make a separate issue for it.